<?xml version="1.0"?>
<robot name ="hexa_bot">

    <material name="blue">
        <color rgba="0 0 0.8 1"/>
    </material>

    <material name="white">
        <color rgba="1 1 1 1"/>
    </material>

    <!-- body -->

    <link name="body">
        <visual>
            <geometry>
                <mesh filename="package://project_demo/meshes/bodyC.stl" scale="0.001 0.001 0.001"/>
            </geometry>
            <material name="blue"/>
        </visual>
    </link>

    <!-- leg1 -->

    <link name="tigh">
        <visual>
            <geometry>
                <mesh filename="package://project_demo/meshes/tighC.stl" scale="0.001 0.001 0.001"/>
            </geometry>
            <material name="white"/>
        </visual>
    </link>

    <link name="knee">
        <visual>
            <geometry>
                <mesh filename="package://project_demo/meshes/calfC.stl" scale="0.001 0.001 0.001"/>
            </geometry>
            <material name="white"/>
        </visual>
    </link>

    <link name="foot">
        <visual>
            <geometry>
                <mesh filename="package://project_demo/meshes/footC.stl" scale="0.001 0.001 0.001"/>
            </geometry>
            <material name="white"/>
        </visual>
    </link>

    <!-- leg 2 -->

    <link name="tigh0">
        <visual>
            <geometry>
                <mesh filename="package://project_demo/meshes/tighC.stl" scale="0.001 0.001 0.001"/>
            </geometry>
            <material name="blue"/>
        </visual>
    </link>
    <link name="knee0">
        <visual>
            <geometry>
                <mesh filename="package://project_demo/meshes/calfC.stl" scale="0.001 0.001 0.001"/>
            </geometry>
            <material name="blue"/>
        </visual>
    </link>

    <link name="foot0">
        <visual>
            <geometry>
                <mesh filename="package://project_demo/meshes/footC.stl" scale="0.001 0.001 0.001"/>
            </geometry>
            <material name="blue"/>
        </visual>
    </link>

    <!-- leg3 -->

    <link name="tigh1">
        <visual>
            <geometry>
                <mesh filename="package://project_demo/meshes/tighC.stl" scale="0.001 0.001 0.001"/>
            </geometry>
            <material name="blue"/>
        </visual>
    </link>

    <link name="knee1">
        <visual>
            <geometry>
                <mesh filename="package://project_demo/meshes/calfC.stl" scale="0.001 0.001 0.001"/>
            </geometry>
            <material name="blue"/>
        </visual>
    </link>

    <link name="foot1">
        <visual>
            <geometry>
                <mesh filename="package://project_demo/meshes/footC.stl" scale="0.001 0.001 0.001"/>
            </geometry>
            <material name="blue"/>
        </visual>
    </link>

    <!-- leg4 -->

    <link name="tigh2">
        <visual>
            <geometry>
                <mesh filename="package://project_demo/meshes/tighC.stl" scale="0.001 0.001 0.001"/>
            </geometry>
            <material name="white"/>
        </visual>
    </link>

    <link name="knee2">
        <visual>
            <geometry>
                <mesh filename="package://project_demo/meshes/calfC.stl" scale="0.001 0.001 0.001"/>
            </geometry>
            <material name="white"/>
        </visual>
    </link>
    <link name="foot2">
        <visual>
            <geometry>
                <mesh filename="package://project_demo/meshes/footC.stl" scale="0.001 0.001 0.001"/>
            </geometry>
            <material name="white"/>
        </visual>
    </link>

    <!-- leg5 -->

    <link name="tigh3">
        <visual>
            <geometry>
                <mesh filename="package://project_demo/meshes/tighC.stl" scale="0.001 0.001 0.001"/>
            </geometry>
            <material name="blue"/>
        </visual>
    </link>

    <link name="knee3">
        <visual>
            <geometry>
                <mesh filename="package://project_demo/meshes/calfC.stl" scale="0.001 0.001 0.001"/>
            </geometry>
            <material name="blue"/>
        </visual>
    </link>

    <link name="foot3">
        <visual>
            <geometry>
                <mesh filename="package://project_demo/meshes/footC.stl" scale="0.001 0.001 0.001"/>
            </geometry>
            <material name="blue"/>
        </visual>
    </link>


    <!-- leg6 -->

    <link name="tigh4">
        <visual>
            <geometry>
                <mesh filename="package://project_demo/meshes/tighC.stl" scale="0.001 0.001 0.001"/>
            </geometry>
            <material name="blue"/>
        </visual>
    </link>

    <link name="knee4">
        <visual>
            <geometry>
                <mesh filename="package://project_demo/meshes/calfC.stl" scale="0.001 0.001 0.001"/>
            </geometry>
            <material name="blue"/>
        </visual>
    </link>
    <link name="foot4">
        <visual>
            <geometry>
                <mesh filename="package://project_demo/meshes/footC.stl" scale="0.001 0.001 0.001"/>
            </geometry>
            <material name="blue"/>
        </visual>
    </link>

    <!-- leg1 -->

    <joint name="tigh_joint" type="continuous">
        <parent link="body"/>
        <child link="tigh"/>
        <origin xyz="0.035 -0.045 -0.006" rpy="0 0 0"/>
        <axis xyz="0 0 1"/>
    </joint>

    <joint name="knee_joint" type="continuous">
        <parent link="tigh"/>
        <child link="knee"/>
        <origin xyz="0.045 0 0" rpy="0 0 0"/>
        <axis xyz="0 1 0"/>
    </joint>

    <joint name="foot_joint" type="continuous">
        <parent link="knee"/>
        <child link="foot"/>
        <origin xyz="0.07 0 0" rpy="0 1 0"/>
        <axis xyz="0 1 0"/>
    </joint>

    <!-- leg2 -->

    <joint name="tigh_joint0" type="continuous">
        <parent link="body"/>
        <child link="tigh0"/>
        <origin xyz="0.035 0.0075 -0.006" rpy="0 0 0"/>
        <axis xyz="0 0 1"/>
    </joint>

    <joint name="knee_joint0" type="continuous">
        <parent link="tigh0"/>
        <child link="knee0"/>
        <origin xyz="0.045 0 0" rpy="0 0 0"/>
        <axis xyz="0 1 0"/>
    </joint>

    <joint name="foot_joint0" type="continuous">
        <parent link="knee0"/>
        <child link="foot0"/>
        <origin xyz="0.07 0 0" rpy="0 1 0"/>
        <axis xyz="0 1 0"/>
    </joint>

    <!-- leg3 -->

    <joint name="tigh_joint1" type="continuous">
        <parent link="body"/>
        <child link="tigh1"/>
        <origin xyz="0.035 0.068 -0.006" rpy="0 0 0"/>
        <axis xyz="0 0 1"/>
    </joint>

    <joint name="knee_joint1" type="continuous">
        <parent link="tigh1"/>
        <child link="knee1"/>
        <origin xyz="0.045 0 0" rpy="0 0 0"/>
        <axis xyz="0 1 0"/>
    </joint>

    <joint name="foot_joint1" type="continuous">
        <parent link="knee1"/>
        <child link="foot1"/>
        <origin xyz="0.07 0 0" rpy="0 1 0"/>
        <axis xyz="0 1 0"/>
    </joint>

    <!-- leg4 front right-->

    <joint name="tigh_joint2" type="continuous">
        <parent link="body"/>
        <child link="tigh2"/>
        <origin xyz="-0.035 -0.045 -0.006" rpy="0 0 3.14"/>
        <axis xyz="0 0 1"/>
    </joint>

    <joint name="knee_joint2" type="continuous">
        <parent link="tigh2"/>
        <child link="knee2"/>
        <origin xyz="0.045 0 0" rpy="0 0 0"/>
        <axis xyz="0 1 0"/>
    </joint>

    <joint name="foot_joint2" type="continuous">
        <parent link="knee2"/>
        <child link="foot2"/>
        <origin xyz="0.07 0 0" rpy="0 1 0"/>
        <axis xyz="0 1 0"/>
    </joint>

    <!-- leg5 -->

    <joint name="tigh_joint3" type="continuous">
        <parent link="body"/>
        <child link="tigh3"/>
        <origin xyz="-0.035 0.0075 -0.006" rpy="0 0 3.14"/>
        <axis xyz="0 0 1"/>
    </joint>

    <joint name="knee_joint3" type="continuous">
        <parent link="tigh3"/>
        <child link="knee3"/>
        <origin xyz="0.045 0 0" rpy="0 0 0"/>
        <axis xyz="0 1 0"/>
    </joint>

    <joint name="foot_joint3" type="continuous">
        <parent link="knee3"/>
        <child link="foot3"/>
        <origin xyz="0.07 0 0" rpy="0 1 0"/>
        <axis xyz="0 1 0"/>
    </joint>

    <!-- leg6 -->

    <joint name="tigh_joint4" type="continuous">
        <parent link="body"/>
        <child link="tigh4"/>
        <origin xyz="-0.035 0.068 -0.006" rpy="0 0 3.14"/>
        <axis xyz="0 0 1"/>
    </joint>

    <joint name="knee_joint4" type="continuous">
        <parent link="tigh4"/>
        <child link="knee4"/>
        <origin xyz="0.045 0 0" rpy="0 0 0"/>
        <axis xyz="0 1 0"/>
    </joint>

    <joint name="foot_joint4" type="continuous">
        <parent link="knee4"/>
        <child link="foot4"/>
        <origin xyz="0.07 0 0" rpy="0 1 0"/>
        <axis xyz="0 1 0"/>
    </joint>
</robot>